Src/FluentAssertions/Streams/StreamAssertions.cs

```diff
@@ -0,0 +1,414 @@
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> BeReadable(string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .WithExpectation("Expected {context:stream} to be readable{reason}, ")
+                .ForCondition(Subject is not null)
+                .FailWith("but found a <null> Stream.")
+                .Then
+                .ForCondition(Subject.CanRead)
+                .FailWith("but it was not.")
+                .Then
+                .ClearExpectation();
+
+            return new AndConstraint<TAssertions>((TAssertions)this);
+        }
+
+        /// <summary>
+        /// Asserts that the current <see cref="Stream"/> is not readable.
+        /// </summary>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> NotBeReadable(string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .WithExpectation("Expected {context:stream} not to be readable{reason}, ")
+                .ForCondition(Subject is not null)
+                .FailWith("but found a <null> Stream.")
+                .Then
+                .ForCondition(!Subject.CanRead)
+                .FailWith("but it was.")
+                .Then
+                .ClearExpectation();
+
+            return new AndConstraint<TAssertions>((TAssertions)this);
+        }
+
+        /// <summary>
+        /// Asserts that the current <see cref="Stream"/> has the <paramref name="expected"/> position.
+        /// </summary>
+        /// <param name="expected">The expected position of the current stream.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> HavePosition(int expected, string because = "", params object[] becauseArgs)
```

Comment:
🤔 Don't you want to make the position a `long`, like the stream's `Position` property?

Reply:
Changed, side effect is the added `L` in the error messages